WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -69,6 +62,15 @@ def index
       @competitions = @competitions.select { |competition| competition.contains?(params[:search]) }
     end
 
+    unless params[:event_ids].empty?
+      @competitions = @competitions.select do |competition|
+        (competition.eventSpecs.split & params[:event_ids]).any?
+      end.sort_by do |competition|
+        -(competition.eventSpecs.split & params[:event_ids]).count
```

Comment:
Just thinking out loud here: I'm not in love with how we have to recompute the number of matching events here. What about something like:

```ruby
required_events = params[:event_ids].map { |event_id| Event.find(event_id) }
...
competitions_and_matching_event_count = @competitions.map { |competition| [ competition, competition.events & required_events ] }
competitions_and_matching_event_count.select { |competition_and_matching_event_count| competition_and_matching_event_count][0] }
competitions_and_matching_event_count.sort_by { |competition_and_matching_event_counth| competition_and_matching_event_couth[1] }
```